Failure to build from source with GOOS=windows

`github.com/prometheus/client_golang v1.1.0` in `go.mod` should be updated to at least `v1.2.0` to fix https://github.com/prometheus/client_golang/issues/642

```
$ GOOS=windows make otelcol
go: downloading github.com/inconshreveable/mousetrap v1.0.0
GO111MODULE=on CGO_ENABLED=0 go build -o ./bin/windows/otelcol -ldflags "-X github.com/open-telemetry/opentelemetry-collector/internal/version.GitHash=ea76156 " ./cmd/otelcol
# github.com/prometheus/client_golang/prometheus
../go/pkg/mod/github.com/prometheus/client_golang@v1.1.0/prometheus/process_collector_windows.go:78:9: assignment mismatch: 2 variables but "golang.org/x/sys/windows".GetCurrentProcess returns 1 values
make: *** [otelcol] Error 2
```

--- a/Makefile
+++ b/Makefile
@@ -14,6 +14,7 @@ GOTEST_OPT?= -race -timeout 30s
 GOTEST_OPT_WITH_COVERAGE = $(GOTEST_OPT) -coverprofile=coverage.txt -covermode=atomic
 GOTEST=go test
 GOOS=$(shell go env GOOS)
+GOARCH=$(shell go env GOARCH)
 ADDLICENCESE= addlicense
 MISSPELL=misspell -error
 MISSPELL_CORRECTION=misspell -w
@@ -54,7 +55,7 @@ benchmark:
 	$(GOTEST) -bench=. -run=notests $(ALL_PKGS)
 
 .PHONY: ci
-ci: all test-with-cover
+ci: all binaries-all-sys test-with-cover
 	$(MAKE) -C testbed install-tools
 	$(MAKE) -C testbed runtests
 
@@ -125,12 +126,12 @@ install-tools:
 
 .PHONY: otelcol
 otelcol:
-	GO111MODULE=on CGO_ENABLED=0 go build -o ./bin/$(GOOS)/otelcol $(BUILD_INFO) ./cmd/otelcol
+	GO111MODULE=on CGO_ENABLED=0 go build -o ./bin/$(GOOS)_$(GOARCH)/otelcol $(BUILD_INFO) ./cmd/otelcol
 
 .PHONY: docker-component # Not intended to be used directly
 docker-component: check-component
 	GOOS=linux $(MAKE) $(COMPONENT)
-	cp ./bin/linux/$(COMPONENT) ./cmd/$(COMPONENT)/
+	cp ./bin/linux_amd64/$(COMPONENT) ./cmd/$(COMPONENT)/
 	docker build -t $(COMPONENT) ./cmd/$(COMPONENT)/
 	rm ./cmd/$(COMPONENT)/$(COMPONENT)
 
@@ -149,6 +150,7 @@ binaries: otelcol
 
 .PHONY: binaries-all-sys
 binaries-all-sys:
-	GOOS=darwin $(MAKE) binaries
-	GOOS=linux $(MAKE) binaries
-	GOOS=windows $(MAKE) binaries
+	GOOS=darwin GOARCH=amd64 $(MAKE) binaries
+	GOOS=linux GOARCH=amd64 $(MAKE) binaries
+	GOOS=linux GOARCH=arm64 $(MAKE) binaries
+	GOOS=windows GOARCH=amd64 $(MAKE) binaries

--- a/testbed/Makefile
+++ b/testbed/Makefile
@@ -5,6 +5,7 @@ GOFMT=gofmt
 GOLINT=golint
 GOVET=go vet
 GOOS=$(shell go env GOOS)
+GOARCH=$(shell go env GOARCH)
 
 .DEFAULT_GOAL := fmt-vet-lint-test
 

--- a/testbed/testbed/test_bed.go
+++ b/testbed/testbed/test_bed.go
@@ -76,9 +76,11 @@ func LoadConfig() error {
 	}
 
 	templateVars := struct {
-		GOOS string
+		GOOS   string
+		GOARCH string
 	}{
-		GOOS: runtime.GOOS,
+		GOOS:   runtime.GOOS,
+		GOARCH: runtime.GOARCH,
 	}
 	var buf bytes.Buffer
 	if err := cfgTemplate.Execute(&buf, templateVars); err != nil {

--- a/testbed/tests/local.yaml
+++ b/testbed/tests/local.yaml
@@ -1,1 +1,1 @@
-agent: ../../bin/{{.GOOS}}/otelcol
+agent: ../../bin/{{.GOOS}}_{{.GOARCH}}/otelcol

